### ISSUE TITLE
HD Assets and Asset Guidelines

### ISSUE THREAD

**Issue by nitramr** [flareteam/flare-game#791]
Hi all,
I'm thinking about some guidelines to keep the asset modding part much more flexible. Currently the game resolution is not state of the art, because it looks very blurry on HD or 4k screens, etc.

I know that in 2013 Clint Bellenger started an HD core and also provided asset files which can render in higher resolutions. https://github.com/clintbellanger/flare-hdcore
But every modder knows how big the effort is to recreate all existing assets in a new resolution. Also for every new game version the assets going more and more and the effort to create a HD core grows too.

My proposal is that in future all assets should be created in HD resolution, like 2x, 4x etc. and can be scaled down to the current resolution to use in the game. In the meanwhile the old assets can migrate step by step to a higher resolution and finally the HD core can be the standard resolution or just as an option like the low resolution assets for mobile games.

Right now I already created new HD assets for fantasy core which I would like to share with you. Especial for the user interface graphics I had to recreate all gimp files because just an up scale don't improve the quality. I think it makes also sense to use a scalable model like SVG or blender files for creating such user interface images. Otherwise we will have the same problem again to create files in more higher resolution. BTW does the engine supports SVG natively like PNG? Maybe it makes sense to use it directly for icons or the interface itself.

In summary:
- we need an update of guidelines to create game assets
- new assets should be initialy created in highes resolution to scaled it down easily for "low resolution" mods

What do you think about that proposals?

Here is the topic of initial HD core discussion. https://github.com/flareteam/flare-game/issues/209

**Comment by dorkster** [comment]
With the 2D assets (i.e. the GUI), we worked with an [existing set of assets](https://opengameart.org/content/rpg-gui-construction-kit-v10), so we were limited to its resolution. Supporting higher resolutions would require starting from the ground up, either we try to recreate the existing, or we adopt a new style like Clint is doing with the assets commissioned for [Wandercall](https://github.com/clintbellanger/wandercall-ch1). Not sure which would be better...

SDL_image only supports [various raster formats](https://www.libsdl.org/projects/SDL_image/docs/SDL_image.html#SEC2), so no SVG. We wouldn't want to use SVG in our case either, since the GUI is very "textured", and SVG lends itself better to simpler designs.

3D assets, on the other hand, are easier to scale up. I think even the existing fantasycore assets could be acceptable rendered at 2x size. The 3D assets only look bad at a certain point if they are poorly textured and/or have a low poly count. Clint's newer models (boss chest, potions, melee weapons) are all very high quality in these regards. Forum user 'rubberduck' has also been working on some [impressive stuff](https://opengameart.org/forumtopic/new-potential-art-for-flare) that's designed to be *at least* as good looking at 2x fantasycore size, if not more.

**Comment by nitramr** [comment]
Hi Dorkster,

thank you for your answer. I didn't know that there is already such big progress for an HD project. The first link leads exactly to the HD resource which saves a lot of time to make a new HD interface for the fantasy core mod. Thanks a lot.

I agree with you about SVG. Also if someone uses SVG files as source to create the interface images it would be easy to export scaled images too.

**Comment by ate088** [comment]
Just curious, but making everything 2x bigger means that the sprite sheets can get pretty big. Is there some sort of upper limit as to how big a sprite sheet can be? eg 2048x2048

**Comment by dorkster** [comment]
@ate088 There's no limit imposed by us. Depending on the renderer, however, there can be hardware limitations. The software sdl renderer generally has no limit besides being able to fit the uncompressed texture in system ram. The hardware sdl renderer is limited by the video card's maximum texture size. So, 2048x2048 is an example limit that would befit some older integrated Intel chips. I added the ability to split tilesets into multiple images in Flare 1.05. A similar refactor/feature should be planned for sprite animations.
